animation_nodes/algorithms/mesh_generation/find_shortest_path.pyx

```diff
@@ -0,0 +1,113 @@
+from ... data_structures import GPStroke
+from ... data_structures cimport (
+    Mesh,
+    LongList,
+    ColorList,
+    FloatList,
+    PolySpline,
+    DoubleList,
+    BooleanList,
+    Vector3DList,
+)
+
+from libc.math cimport INFINITY
+from . line import getLinesMesh
+from ... math cimport distanceVec3
+
+# Dijkstra's algorithm (https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm) but is implemented
+# such a way to handle multi-sources and mesh with multiple islands.
+def getShortestPath(Mesh mesh, LongList sources, Py_ssize_t target):
+    cdef LongList previousVertices = computeShortestPathTree(mesh, sources, target)
+    cdef LongList indices
+    cdef Py_ssize_t index
+
+    indices = LongList()
+    index = target
+    if previousVertices.data[index] == -1: return indices
+    while index != -1:
+        indices.append(index)
+        index = previousVertices.data[index]
+
+    return indices.reversed()
+
+def getShortestTree(Mesh mesh, LongList sources, str pathType):
+    cdef Vector3DList vertices = mesh.vertices
+    cdef long vertexCount = vertices.length
+    cdef LongList previousVertices = LongList(length = vertexCount)
```

Comment:
This is redundant. You are allocating here and another time in in the function below.

Reply:
But we need *Vertices* to construct the paths. Do I make the change in the `computeShortestPathTree` method so that it also returns *Vertices* as well?

Reply:
I mean `LongList(length = vertexCount)` is reduandant. It will be overwritten anyway in the next line.

Reply:
Okay.